src/mono/wasm/runtime/binding_support.js

```diff
@@ -2043,6 +2071,57 @@ var BindingSupportLib = {
 		return BINDING.js_to_mono_obj (res)
 	},
 
+	mono_wasm_add_event_listener: function (objHandle, name, listenerId, optionsHandle) {
+		var nameRoot = MONO.mono_wasm_new_root (name);
+		try {
+			BINDING.bindings_lazy_init ();
+			var obj = BINDING.mono_wasm_require_handle(objHandle);
+			if (!obj)
+				throw new Error("Invalid JS object handle");
+			var listener = BINDING._get_weak_delegate_from_handle(listenerId);
+			if (!listener)
+				throw new Error("Invalid listener ID");
+			var sName = BINDING.conv_string(nameRoot.value);
+
+			var options = optionsHandle
+				? BINDING.mono_wasm_require_handle(optionsHandle)
+				: null;
+			// console.log(`${obj}.addEventListener(${sName}, ${listener}, ${options})`);
+
+			if (options)
+				obj.addEventListener(sName, listener, options);
+			else
+				obj.addEventListener(sName, listener);
+			return 0;
+		} catch (exc) {
+			return BINDING.js_string_to_mono_string(exc.message);
+		} finally {
+			nameRoot.release();
+		}
+	},
+
+	mono_wasm_remove_event_listener: function (objHandle, name, listenerId, capture) {
```

Comment:
this method should also remove it from `_weak_delegate_registry` I think

Reply:
Nope, it could be registered to multiple objects

Reply:
Could you please document this design decision in the code comment ?